2016/proposals/stolliker-ryan.md

```diff
@@ -0,0 +1,67 @@
+# Ryan Stolliker: Result Aggregation Server for Software Carpentry
+
+## Abstract
+
+During Software Carpentry's workshops which teach computing skills to the greater scientific community, learners use Python scripts to test that they have installed the programming environment and other dependencies correctly. Currently, this information is known only to the participants and cannot be used or tracked by the organizers. Thus, the goal of this project is to enable the collection of this installation data and to store it for future use and analysis.
+
+## Technical Details
+
+### Server
+
+All server management will be written in Python. The server itself will be implemented using [bottle](http://bottlepy.org/docs/dev/index.html), a simple web framework with functionality to parse forms and JSON requests and responses. SQLite will be used to create and manage the database which holds the results.
```

Comment:
Django is perhaps a stronger framework overall, but I believe that for creating an API, a more minimalist framework is better. Between Flask and bottle, it is fairly even and some of the ways they approach doing things are very similar. I tend to prefer bottle over Flask because a few years ago Flask did not play well with Python 3, but these problems have largely been fixed now, and I also like the way bottle handles JSON and forms requests. To be honest I’m not devoted to one particular framework and would be willing to switch if necessary.